projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts

```diff
@@ -86,4 +93,32 @@ describe('ReportBPComponent', () => {
     await systolicInputHarness.setValue('252');
     expect(systolicFormField?.errors?.['max']).toBeTruthy();
   });
+
+  it('should submit prioer blood pressure Form', async () => {
+    fhirService.createResourceData.and.callThrough().and.returnValue({ resource: 'resource' });
+    fhirService.addPatientData.and.callThrough();
+    spyOn(component, 'onSubmit').and.callThrough();
+    component.onSubmit();
+    expect(component.onSubmit).toHaveBeenCalled();
```

Comment:
there's no point in checking whether onSubmit() has been called when you just called it yourself on the previous line.